test/e2e/proxy_test.go

```diff
@@ -99,3 +138,75 @@ func (tc *testContext) waitForValidTrustedCAConfigMap() error {
 	}
 	return nil
 }
+
+// getSystemEnvVar returns the value corresponding to the input system ENV var as set in the registry
+func (tc *testContext) getSystemEnvVar(addr, variableName string) (string, error) {
+	command := fmt.Sprintf("Get-ChildItem -Path Env: | Where-Object -Property Name -eq '%s' | Format-List ",
+		variableName)
+	out, err := tc.runPowerShellSSHJob(fmt.Sprintf("get-%s-system-env-var", strings.ToLower(variableName)),
+		command, addr)
+	if err != nil {
+		return "", fmt.Errorf("error running SSH job: %w", err)
+	}
+	lines := strings.Split(out, "\n")
+	var valueLines []string
+	var value string
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if strings.HasPrefix(line, "Value") {
+			parts := strings.SplitN(line, ":", 2)
+			if len(parts) == 2 {
+				valueLine := strings.TrimSpace(strings.TrimPrefix(parts[1], "Value:"))
+				valueLines = []string{valueLine}
+			} // used when a long value is split into multiple lines
+		} else if line != "" {
+			valueLines = append(valueLines, line)
+		}
+		if len(valueLines) > 0 {
+			value = strings.Join(valueLines, "")
+			value = strings.ReplaceAll(value, ";", ",")
+		}
+	}
+	return value, nil
+}
+
+// getServiceProxyEnvVars returns a map of proxy environment variables present in a service's config
```

Comment:
I don't follow this. Where are we adding the proxy variables to the service's configuration?